src/mintpy/prep_nisar.py

```diff
@@ -263,20 +274,47 @@ def read_and_interpolate_geometry(gunw_file, dem_file, xybbox):
     # dem_subset_array = np.zeros((subset_rows, subset_cols), dtype=raster_array.dtype)
     Y_2d, X_2d = np.meshgrid(ycoord, xcoord, indexing='ij')
 
-    if not src_epsg == dst_epsg:
-        coord_transform = Transformer.from_crs(dst_epsg, src_epsg, always_xy=True)
+    if not dem_src_epsg == dst_epsg:
+        coord_transform = Transformer.from_crs(dst_epsg, dem_src_epsg, always_xy=True)
         x_dem, y_dem = coord_transform.transform(X_2d.flatten(), Y_2d.flatten())
     else:
         x_dem, y_dem = X_2d.flatten(), Y_2d.flatten()
 
     cols = ((y_dem - geotransform[3]) / geotransform[5]).astype(int)
     rows = ((x_dem - geotransform[0]) / geotransform[1]).astype(int)
 
-    dem_subset_array = raster_array[cols.reshape(subset_rows, subset_cols), rows.reshape(subset_rows, subset_cols)]
+    dem_subset_array = dem_raster_array[cols.reshape(subset_rows, subset_cols), rows.reshape(subset_rows, subset_cols)]
 
     slant_range, incidence_angle = interpolate_geometry(X_2d, Y_2d, dem_subset_array, rdr_coords)
 
-    return dem_subset_array, slant_range, incidence_angle
+
+    if mask_file in ['auto', 'None', None]:
+        print('*** No mask was found ***')
+        mask_subset_array = np.ones(dem_subset_array.shape, dtype='byte')
+    else:
+        try:
+            mask_dataset = gdal.Open(mask_file, gdal.GA_ReadOnly)
+            geotransform = mask_dataset.GetGeoTransform()
+            proj = gdal.osr.SpatialReference(wkt=mask_dataset.GetProjection())
+            mask_src_epsg = int(proj.GetAttrValue('AUTHORITY', 1))
+            mask_raster_array = mask_dataset.ReadAsArray()
+            del mask_dataset
+
+            if not mask_src_epsg == dst_epsg:
+                coord_transform = Transformer.from_crs(dst_epsg, mask_src_epsg, always_xy=True)
```

Comment:
Similar to my comment above on DEM transformation , I think you could use gdal.warp here as well. Since your mask is binary, you may use nearest neighbor interpolation.